src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -0,0 +1,617 @@
+            this::getChassisSpeeds, // robot relative chassis speeds supplier
+            this::setOutputRobotRelativeSpeeds, // robot relative chassis speeds consumer
+            new PIDConstants(DriveCal.DRIVING_P, DriveCal.DRIVING_I, DriveCal.DRIVING_D), // PID Constants for
+                                                                                          // translation
+            new PIDConstants(DriveCal.TURNING_P, DriveCal.TURNING_I, DriveCal.TURNING_D), // PID Constants for rotation
+            DriveConstants.DRIVE_WHEEL_FREE_SPEED_METERS_PER_SECOND, // maxModuleSpeed
+            DriveConstants.DRIVE_BASE_RADIUS_METERS,
+            new ReplanningConfig(), // creates a path replanning configuration with the default config
+            () -> {
+              return false;
+            }, // boolean supplier for shouldFlipPath
+            this),
+
+        new InstantCommand(
+            () -> {
+              targetHeadingDegrees = getHeadingDegrees();
+            }),
+
+        new PrintCommand("Finished a trajectory"));
+  }
+
+  /** Returns a past pose, given a latency adjustment */
+  public Pose2d getPastPose(double latencySec) {
+    Pose2d curPose = getPose();
+    double latencyAdjustmentSec = 0.00;
+    latencySec += latencyAdjustmentSec;
+    Transform2d pastTransform = new Transform2d(
+        new Translation2d(
+            -lastSetChassisSpeeds.vxMetersPerSecond * latencySec,
+            -lastSetChassisSpeeds.vyMetersPerSecond * latencySec),
+        Rotation2d.fromRadians(lastSetChassisSpeeds.omegaRadiansPerSecond * latencySec)
+            .unaryMinus());
+    Pose2d pastPose = curPose.plus(pastTransform);
+    return pastPose;
+  }
+
+  public void setLimelightTargetFromTransform(Transform2d transform, double latencySec) {
+    // Transform is to get the limelight to the correct location, not to get the
+    // robot
+    // Here we correct for that
+    Transform2d flipTransform = new Transform2d(
+        new Translation2d(-transform.getX(), transform.getY()), transform.getRotation());
+
+    double latencyAdjustmentSec = 0.00;
+
+    Pose2d curPose = getPose();
+    Pose2d pastPose = getPastPose(latencyAdjustmentSec);
+
+    final boolean useLatencyAdjustment = true;
+
+    targetPose = useLatencyAdjustment
+        ? Optional.of(pastPose.plus(flipTransform))
+        : Optional.of(curPose.plus(flipTransform));
+  }
+
+  public PathPlannerPath pathToPoint(Pose2d finalPose, double finalSpeedMetersPerSec) {
+    Pose2d curPose = getPose();
+    Transform2d finalTransform = new Transform2d(finalPose.getTranslation(), finalPose.getRotation());
+    System.out.println(
+        "Trajectory Transform: " + finalTransform.getX() + " " + finalTransform.getY());
+    Rotation2d finalHeading = Rotation2d.fromDegrees(180);
+    Rotation2d finalHolonomicRotation = finalPose.getRotation();
+
+    PathPlannerPath path = PathPlannerPath.fromPathPoints(Arrays.asList(new PathPoint(finalTransform.getTranslation(),
```

Comment:
`fromPathPoints` is for a path that's already smoothed. I think we need to use a different function for path planner to smooth the path for us